### PR TITLE
Add tests for TaskController

### PR DESCRIPTION
**Pull request by JoseCastroA**
# :bulb: Type of change
- [ ] bugfix :bug:.
- [x] new feature.
- [ ] refactoring.

# :notebook: Description
In this PR I added basic tests for TaskController

# :bomb: How Has This Been Tested?

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -31,13 +31,16 @@ gem 'redis', '~> 4.0'
 gem 'bootsnap', '>= 1.4.4', require: false
 
 group :test do
+  gem 'factory_bot'
   gem 'rspec-rails', '~> 4.1.0'
   gem 'simplecov', require: false
+  gem 'rails-controller-testing'
```
Comment (brayvasq) [inline Gemfile]: @JoseCastroA What's the purpose of this gem?

File: spec/controllers/api/v1/tasks_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+RSpec.describe Api::V1::TasksController, type: :controller do
+  let(:tasks) { create_list(:task, 10) }
+
+  # index action
+  describe 'GET index' do
+    subject { get :index, format: :json }
+
+    it 'status should be' do
+      expect(subject.status).to eq(200)
```
Comment (brayvasq) [inline spec/controllers/api/v1/tasks_controller_spec.rb]: @JoseCastroA I think this kind of tests could be written as follows:
```ruby
it { is_expected.to have_http_status(:ok) }
```

File: spec/controllers/api/v1/tasks_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    end
+
+    it 'when there are not tasks' do
```
Comment (brayvasq) [inline spec/controllers/api/v1/tasks_controller_spec.rb]: @JoseCastroA I think this example should be in a different context, something like the following
```ruby
describe 'GET index' do
  context 'when there are not tasks' do
    # this example.
  end

  context 'when there are tasks' do
    let!(:tasks) { .... }

    # Examples for success case.
  end
end
```

File: spec/controllers/api/v1/tasks_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      expect(subject.body).to eq(Task.all.to_json)
+    end
+  end
+
+  # create action
+  describe 'POST #create' do
+    subject { post :create, format: :json, params: params }
+
+    context 'when params are correct' do
+      let(:params) { { task: { title: Faker::Hacker.say_something_smart, description: Faker::Hacker.adjective, time: Faker::Number.number(digits: 3) } } }
```
Comment (brayvasq) [inline spec/controllers/api/v1/tasks_controller_spec.rb]: @JoseCastroA This line is too long.

**Review by brayvasq — CHANGES_REQUESTED** [review]

**Comment by JoseCastroA** [comment]
@brayvasq Changes are ready

File: spec/controllers/api/v1/tasks_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  # update action
+  describe 'PATCH #update' do
+    subject { put :update, format: :json, params: params }
+
+    context 'when user exist in database' do
+      let(:task) { create(:task) }
+      let(:params) { { id: task.id, task: { title: 'New name' } } }
+
+      context 'when data is provided is valid' do
```
Comment (brayvasq) [inline spec/controllers/api/v1/tasks_controller_spec.rb]: @JoseCastroA I think we don't need this context.

File: spec/controllers/api/v1/tasks_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    subject { get :index, format: :json }
+
+    it { is_expected.to have_http_status(:ok) }
+
+    context 'when there are not tasks' do
+      it { expect(subject.body).to eq('[]') }
+    end
+
+    context 'when there are tasks' do
+      let!(:tasks) { create_list(:task, 10) }
```
Comment (brayvasq) [inline spec/controllers/api/v1/tasks_controller_spec.rb:15]: **praise:** Good use of `let!`

**Review by brayvasq — CHANGES_REQUESTED** [review]

**Comment by JoseCastroArias** [comment]
@brayvasq Changes are ready.

**Review by brayvasq — APPROVED** [review]

**Pull request merged**